### ISSUE TITLE
Support non-utf8 path?

### ISSUE THREAD

**Issue by WindSoilder** [olson-sean-k/wax#26]
Refer to: https://github.com/nushell/nushell/issues/2987
If we create a file from the code:
```rust
  use std::ffi::OsString;
  use std::fs;
  use std::io;
  use std::os::unix::ffi::OsStringExt;
  
  fn main() -> io::Result<()> {
      let chars: Vec<u8> = (1..=46)
          .into_iter()
          .chain(48..=70)
          .chain(150..200)
          .collect();
  
      fs::create_dir(OsString::from_vec(chars.clone()))
  }
```

And run glob walk with the following code:
```rust
use wax::Glob as WaxGlob;

fn main() {
    let glob = WaxGlob::new("*").unwrap();
    let glob_results: Vec<_> = glob.walk(".").collect();
}
```

It makes the program panic, with the message says that: unexpected encoding: Utf8Error

**Comment by olson-sean-k** [comment]
I just took a look at nushell/nushell#2987. Is the `ls` command in this context using `Glob::walk`?. Since there is no reported panic, it seems like it is not, but I'm not sure. If it is, I'd recommend using the `walkdir` crate instead unless the `ls` command also accepts or otherwise uses glob expressions other than `*` and `**` (or possibly if nushell is already using Wax elsewhere; `Glob::walk` is currently implemented atop `walkdir`).

**Comment by WindSoilder** [comment]
Yeah, that issue doesn't mention about `Glob::walk`.  But nushell has a [glob command](https://github.com/nushell/nushell/blob/main/crates/nu-command/src/filesystem/glob.rs) which make use of `Glob::walk`.

**Comment by olson-sean-k** [comment]
This has been fixed in 5263d5a. That change removes the `bstr` dependency and unconditionally uses `OsStr::to_string_lossy` when constructing a `CandidatePath` from an `OsStr` or `Path`.
